Imabulary/Imabulary
# IMB-336: Get rid of the Navigator for the basic navigation

## Overview

- Avoid using Navigator for navigation between screens
- Use the Tabbar only on the Home screen but not for the nested ones

## Test cases

- Check dialogs
- Change navigation from Home or Profile screens

## Screenshots

![Simulator Screenshot - iPhone 15 Pro Max - 2024-11-13 at 19 23 15](https://github.com/user-attachments/assets/c9d86fe8-d21e-464b-8bd2-257322238075)
![Simulator Screenshot - iPhone 15 Pro Max - 2024-11-13 at 19 23 12](https://github.com/user-attachments/assets/f3390388-aa6f-40ae-89b7-23fc4b782b01)
![Simulator Screenshot - iPhone 15 Pro Max - 2024-11-13 at 19 23 10](https://github.com/user-attachments/assets/401ff26d-5f17-4853-8ad5-98d508718f5b)

https://github.com/user-attachments/assets/41d29afc-ea72-47fb-b40a-1bb22f238a06



## Issue

[IMB-308](https://www.notion.so/Migrate-to-app_route-instead-of-standard-Flutter-s-navigation-12575429ec6580128777c3a373f59666?pvs=4)
[IMB-336](https://www.notion.so/FE-Implement-new-router-13475429ec6580faa23aeb68f8aeb202?pvs=4)
